src/scripts/themeManager.js

```diff
@@ -17,10 +17,18 @@ function getThemes() {
 
 function getThemeStylesheetInfo(id) {
     return getThemes().then(themes => {
-        const theme = themes.find(theme => {
+        let theme = themes.find(theme => {
             return id ? theme.id === id : theme.default;
         });
 
+        if (!theme) {
+            theme = {
+                'name': 'Dark',
+                'id': 'dark',
+                'default': true
+            };
+        }
+
```

Comment:
This should probably be fixed in the web settings rather than here.

Reply:
I think this is to handle the case where an invalid id is passed to this function. Instead of being `null` it defaults to dark. I don't think it could be handled anywhere else really.

Reply:
Then it is probably better to make sure `themes` array is valid in `webSettings` and search for default theme
```js
theme = themes.find(theme => {
    return theme.default;
});
```

Reply:
i passed the default theme to webSettings and guarantee that there's a default theme when you call getThemes

Reply:
To handle what @thornbill said (invalid id) we need to search for default theme after this one
https://github.com/jellyfin/jellyfin-web/blob/2c899c93dff42d7cefc1c2cbeb0e63eaf5d02b8b/src/scripts/themeManager.js#L20-L22
or even rewrite it like this
```js
let theme;
if (id) theme = themes.find((theme) => theme.id === id);
if (!theme) theme = themes.find((theme) => theme.default);
```

Reply:
i verify that in webSettings when loading the themes. I made a function to search for the existence of a default theme. if such is not present, i inject it in the themes array and return it. So, there's always guarantee that there's a default theme.

Reply:
I saw that. The problem will be, for example, if `id` is theme that has been removed - we won't find it in the array.

Reply:
Oh i see now. I changed the approach and created a getDefaultTheme in webSettings. When loading the themes, i still call the checkDefaultTheme, but this will only set an internal prop for a default Theme.

Then in themeManager you only need to iterate when id exists, and if theme is undefined, will set it with the default one. 

Please check my changes if they are ok for you.

Reply:
Yes, that works :+1: